src/components/TeamTab.astro

```diff
@@ -31,16 +34,28 @@ const currentPath = Astro.url.pathname;
 <div class="tab">
   {
     tabItems.map((item, index) => (
-      <a
-        class:list={{ active: currentPath === item.link }}
-        href={currentPath === item.link ? "#team" : `${item.link}#team`}
-      >
+      <a href="#team" id={item.team}>
         <div class="tab-title">{item.displayName}</div>
       </a>
     ))
   }
 </div>
 
+<script>
+  import { teamTableState } from "../utils/state.ts";
+
+  // remove repetition here
+  document.getElementById("ctms").addEventListener('click', () => {
+      teamTableState.teamState = "ctms";
+  });
+  document.getElementById("executives").addEventListener('click', () => {
+      teamTableState.teamState = "executives";
```

Comment:
Does this work btw? Can it not be moved inside a react component and handle all reactivity there?

Reply:
Is it possible to listen for events from a different component without rewriting this entire component in react?